### PR TITLE
Refactor filename handling

### PR DESCRIPTION
**Pull request by nafur**
This PR simplifies how we store the current input file name and handle setting and getting it.
It is now an option, that can also be set and get via `setInfo()` and `getInfo()`.

### CODE REVIEW

File: src/util/statistics_public.cpp [hunk 1]
```diff
@@ -30,7 +30,6 @@ void registerPublicStatistics(StatisticsRegistry& reg)
   reg.registerHistogram<api::Kind>("api::TERM", false);
 
   reg.registerValue<std::string>("driver::filename", false);
-  reg.registerValue<std::string>("driver::sat/unsat", false);
```
Comment (4tXJ7f) [inline src/util/statistics_public.cpp:33]: This change looks unrelated.

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
LGTM except for one minor comment.

File: src/smt/smt_engine.cpp [hunk 2]
```diff
@@ -432,7 +413,10 @@ void SmtEngine::setInfo(const std::string& key, const std::string& value)
 
   if (key == "filename")
   {
-    d_env->setFilename(value);
+    d_env->d_options.driver.filename = value;
```
Comment (4tXJ7f) [inline src/smt/smt_engine.cpp:416]: Can we remove the filename and the setters/getters from `Env` now?

File: src/util/statistics_public.cpp (hunk 1 above)
Comment (4tXJ7f) [reply src/util/statistics_public.cpp:33]: Discussed offline. The change is on purpose and this is fine with me.

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the updates.

File: src/util/statistics_public.cpp (hunk 1 above)
Comment (nafur) [reply src/util/statistics_public.cpp:33]: It technically is, but fits well with this PR
